xbmc/rendering/dx/DeviceResources.cpp

```diff
@@ -281,8 +282,8 @@ void DX::DeviceResources::CreateDeviceResources()
   ComPtr<ID3D11DeviceContext> context;
 
   HRESULT hr = D3D11CreateDevice(
-      nullptr,                   // Specify nullptr to use the default adapter.
-      D3D_DRIVER_TYPE_HARDWARE,  // Create a device using the hardware graphics driver.
+      m_adapter.Get(),           // Specify nullptr to use the default adapter.
+      D3D_DRIVER_TYPE_UNKNOWN,   // Create a device using the hardware graphics driver.
```

Comment:
D3D_DRIVER_TYPE value have to depends on `m_adapter` value, `m_adapter` may has null value on uwp

Reply:
ifdef?

Reply:
in runtime `m_adapter != null ? D3D_DRIVER_TYPE_UNKNOWN : D3D_DRIVER_TYPE_HARDWARE`